README.md

```diff
@@ -1,8 +1,262 @@
-# RFC
+# Amethyst RFCs - [Active RFC List TBD]
 
-This is the repository where we work and keep track of the all RFC documents related to Amethyst.
-Having this repository allows people to participate in decision making by opening a pull request with a new document.
+[Amethyst RFCs]:
 
-## Creating a new RFC
+Many changes, including bug fixes and documentation improvements can be
+implemented and reviewed via the normal GitHub pull request workflow.
 
-The specifics of the RFC process are not yet discussed, but will be documented in the first RFC.
+Some changes though are "substantial", and we ask that these be put through a
+bit of a design process and produce a consensus among the Amethyst community and
+the [sub-team]s.
+
+The "RFC" (Request for Comments) process is intended to provide a consistent
+and controlled path for new features to enter the engine and associated
+libraries, so that all stakeholders can be confident about the direction in which
+the engine is evolving.
+
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+  - [Opening] Should we put a link to the Discord server here?
+  - [Table of Contents]
+  - [When you need to follow this process]
+  - [Before creating an RFC]
+  - [What the process is]
+  - [The RFC life-cycle]
+  - [Reviewing RFCs]
+  - [Implementing an RFC]
+  - [RFC Postponement]
+  - [Help this is all too informal!]
+  - [License]
+
+
+## When you need to follow this process
+[When you need to follow this process]: #when-you-need-to-follow-this-process
+
+You need to follow this process if you intend to make "substantial" changes to
+Amethyst, its component crates, or the RFC process itself. The definition of substantial
+is somewhat fluid, and to an extent set by the community. Some examples might include:
+
+  - A change to a different crate used for physics calculations
+  - Changes to how the editor communicates with the engine
+  - Significant protocol changes to the networking library
+  - Anything that has the potential to produce a substantial negative impact on users
+
+Some changes do not require an RFC:
+
+  - Rephrasing, reorganizing, refactoring, or otherwise "changing shape does
+    not change meaning".
+  - Additions that strictly improve objective, numerical quality criteria
+    (warning removal, speedup, better platform coverage, more parallelism, trap
+    more errors, etc.)
+  - Additions only likely to be _noticed by_ other developers-of-amethyst,
+    invisible to users-of-amethyst.
+  - More tests and benchmarks
+
+If you submit a pull request to implement a new feature without going through
+the RFC process, it may be closed with a polite request to submit an RFC first.
```

Comment:
Do we want to be that strict? I can imagine features that

1) don't require much to be changed
2) don't have reasonable alternatives

e.g. new camera component, new light types, or generally adding a variant of something that already exists

Reply:
What happens if we're undecided whether an RFC is required or not?

Reply:
If there is debate, then there should be an RFC. If it was worth it, then good, if it wasn't, the RFC will be merged quickly.

Reply:
Yup, generally if an issue goes into the stalled period then it should open an RFC.

Reply:
We could even have the RFC bot open issue automatically if its in the stalled period for 2 weeks maybe 😄

Reply:
Do we have consensus on stalled issues opening an RFC? @torkleyy @LucioFranco @Moxinilian

Reply:
Hi! I don't think 2 week is enough. Maybe someone is working on code and the rfc was ready to be implemented. In which case, it could easily be stalled for 2+ months before anything new is written on it.

As for pull requests being forced to open an rfc, I would impose that only on very very controversial changes. Like when we added the optionnal sdl dependency for controller support.

Reply:
I'm not sure, I don't think stalled issues are automatically an RFC. RFCs can be seen as decisions that need to be made, and issues get stale for many reasons other than missing decisions.

Reply:
How about something like:

```
If you submit a pull request that ends up being controversial and requires nore
discussion, you may be asked to submit an RFC.
```

Reply:
That's definitely a good notice, although not the only requirement for an RFC. Big PRs should create an RFC beforehand to avoid doing unnecessary work or merging something even though it's not what we'd like to have. It's kind of hard to define what needs an RFC.. I don't think it should be controversial stuff only, since when you call it controversial, it means there needs to be some reviewer who already has other ideas, which might not be the case without an RFC process. At the same time we should not make too much an RFC process since it creates some overhead..

Reply:
This process is not carved onto adamantium tablets by a Wakandan Vibranium laser. =) It can be changed if it needs to be.

This section of the doc, I think, is intended to handle the case of what seems to be a minor feature suddenly turning out to be much bigger or more complex than anyone anticipated. We could say:

```
If you submit a PR that sparks a great deal of discussion or push-back from the community, you may be asked to submit it as a PR. Sometimes what appears to be a minor change can have a far-reaching impact.
```